From 9cc4f4c86e69146e821a55caae286956f7053735 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Tue, 21 Mar 2017 12:08:27 +0800
Subject: [PATCH 1638/5242] MLK-14238-05 ARM: dts: imx: Add PWM backlight
 support on i.mx7ulp evk

commit  0ee5a8df691ec924fd29be9039cf354350d230c0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add PWM backlight support on i.MX7ULP EVK board.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
[Octavian: fix checkpatch warnings]
Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index e616bf4..c4bbf7e 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -33,11 +33,10 @@
 	};
 
 	backlight {
-		compatible = "gpio-backlight";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_backlight>;
-		gpios = <&gpio3 2 GPIO_ACTIVE_HIGH>;
-		default-on;
+		compatible = "pwm-backlight";
+		pwms = <&pwm0 1 5000000>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <6>;
 		status = "okay";
 	};
 
@@ -197,9 +196,9 @@
 			>;
 		};
 
-		pinctrl_backlight: backlight_grp {
+		pinctrl_pwm0: pwm0_grp {
 			fsl,pins = <
-				IMX7ULP_PAD_PTF2__PTF2		0x20100
+				IMX7ULP_PAD_PTF2__TPM4_CH1	0x403
 			>;
 		};
 
@@ -460,6 +459,13 @@
 	status = "disabled";
 };
 
+&pwm0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&pinctrl_pwm0>;
+	pinctrl-1 = <&pinctrl_pwm0>;
+	status = "okay";
+};
+
 &rpmsg{
 	/*
 	 * 64K for one rpmsg instance, default using 2 rpmsg instances:
-- 
1.7.9.5

